The target libtool is used when building host binaries, which predictably
doesn't work.

Upstream-Status: Submitted (https://bugs.freedesktop.org/show_bug.cgi?id=44618)
Signed-off-by: Jonathan Liu <net147@gmail.com>
Signed-off-by: Ross Burton <ross.burton@intel.com>

diff --git a/src/glsl/builtin_compiler/Makefile.am b/src/glsl/builtin_compiler/Makefile.am
index e11a17f..8ebe0a2 100644
--- a/src/glsl/builtin_compiler/Makefile.am
+++ b/src/glsl/builtin_compiler/Makefile.am
@@ -64,6 +64,8 @@ AM_CXXFLAGS = $(AM_CFLAGS)
 include ../Makefile.sources
 
 noinst_PROGRAMS = builtin_compiler
+
+if !CROSS_COMPILING
 noinst_LTLIBRARIES = libglslcore.la libglcpp.la
 
 libglcpp_la_SOURCES =					\
@@ -73,6 +75,7 @@ libglcpp_la_SOURCES =					\
 libglslcore_la_SOURCES =				\
 	$(BUILTIN_COMPILER_GENERATED_CXX_FILES)		\
 	$(LIBGLSL_FILES)
+endif
 
 builtin_compiler_SOURCES = \
 	$(top_srcdir)/src/mesa/main/hash_table.c	\
@@ -81,4 +84,14 @@ builtin_compiler_SOURCES = \
 	$(top_srcdir)/src/mesa/program/symbol_table.c	\
 	$(BUILTIN_COMPILER_CXX_FILES)			\
 	$(GLSL_COMPILER_CXX_FILES)
+
+if CROSS_COMPILING
+builtin_compiler_SOURCES += \
+	$(LIBGLCPP_GENERATED_FILES) \
+	$(LIBGLCPP_FILES) \
+	$(BUILTIN_COMPILER_GENERATED_CXX_FILES) \
+	$(LIBGLSL_FILES)
+builtin_compiler_CPPFLAGS = $(AM_CPPFLAGS)
+else
 builtin_compiler_LDADD = libglslcore.la libglcpp.la
+endif
